core/cmake does not support HTTPS

Hello,

I attempting to compile another project that uses cmake and receiving the below attached error.  

This seems to be because cmake comes with a bundled curl that does not support HTTPS [1]

Thanks
Q 

* [1]  https://stackoverflow.com/a/33512778

```     
src='https://github.com/neovim/libvterm/archive/a9c7c6fd20fa35e0ad3e0e98901ca12dfca9c25c.tar.gz'
     dst='/hab/cache/src/neovim-0.3.0/.deps/build/downloads/libvterm/a9c7c6fd20fa35e0ad3e0e98901ca12dfca9c25c.tar.gz'
     timeout='none'
CMake Error at /hab/cache/src/neovim-0.3.0/third-party/cmake/DownloadAndExtractFile.cmake:74 (message):
  error: downloading
  'https://github.com/neovim/libvterm/archive/a9c7c6fd20fa35e0ad3e0e98901ca12dfca9c25c.tar.gz'
  failed

    status_code: 1
    status_string: "Unsupported protocol"
    log: Protocol "https" not supported or disabled in libcurl

  Closing connection -1
```

--- a/cmake/plan.sh
+++ b/cmake/plan.sh
@@ -11,15 +11,14 @@ pkg_shasum=80d0faad4ab56de07aa21a7fc692c88c4ce6156d42b0579c6962004a70a3218b
 pkg_deps=(
   core/glibc
   core/gcc-libs
+  core/curl
+  core/zlib
 )
 pkg_build_deps=(
   core/coreutils
   core/diffutils
   core/make
   core/gcc
-  core/curl
-  core/zlib
-  core/bzip2
 )
 
 pkg_lib_dirs=(lib)
@@ -27,7 +26,18 @@ pkg_include_dirs=(include)
 pkg_bin_dirs=(bin)
 
 do_build() {
-  ./bootstrap
+  ZLIB=$(pkg_path_for core/zlib)
+  ZLIB_LIB="${ZLIB}/lib"
+  ZLIB_INCLUDE="${ZLIB}/include"
+  CURL=$(pkg_path_for core/curl)
+  CURL_LIB="${CURL}/lib"
+  CURL_INCLUDE="${CURL}/include"
+
+  LD_LIBRARY_PATH=${ZLIB_LIB}:${CURL_LIB}
+  ./bootstrap --parallel="$(nproc)" --system-curl -- -DLD_LIBRARY_PATH=$LD_LIBRARY_PATH \
+    -DZLIB_LIBRARY:FILEPATH=${ZLIB_LIB}/libz.so -DZLIB_INCLUDE_DIR:PATH=${ZLIB_INCLUDE} \
+    -DCURL_LIBRARY:FILEPATH=${CURL_LIB}/libcurl.so  -DCURL_INCLUDE_DIR:PATH=${CURL_INCLUDE}
+
   ./configure --prefix="${pkg_prefix}"
   make -j "$(nproc)"
 }

--- a/cmake/plan.sh
+++ b/cmake/plan.sh
@@ -34,9 +34,9 @@ do_build() {
   CURL_INCLUDE="${CURL}/include"
 
   LD_LIBRARY_PATH=${ZLIB_LIB}:${CURL_LIB}
-  ./bootstrap --parallel="$(nproc)" --system-curl -- -DLD_LIBRARY_PATH=$LD_LIBRARY_PATH \
-    -DZLIB_LIBRARY:FILEPATH=${ZLIB_LIB}/libz.so -DZLIB_INCLUDE_DIR:PATH=${ZLIB_INCLUDE} \
-    -DCURL_LIBRARY:FILEPATH=${CURL_LIB}/libcurl.so  -DCURL_INCLUDE_DIR:PATH=${CURL_INCLUDE}
+  ./bootstrap --parallel="$(nproc)" --system-curl -- -DLD_LIBRARY_PATH="${LD_LIBRARY_PATH}" \
+    -DZLIB_LIBRARY:FILEPATH="${ZLIB_LIB}/libz.so" -DZLIB_INCLUDE_DIR:PATH="${ZLIB_INCLUDE}" \
+    -DCURL_LIBRARY:FILEPATH="${CURL_LIB}/libcurl.so"  -DCURL_INCLUDE_DIR:PATH="${CURL_INCLUDE}"
 
   ./configure --prefix="${pkg_prefix}"
   make -j "$(nproc)"

--- a/cmake/plan.sh
+++ b/cmake/plan.sh
@@ -33,8 +33,7 @@ do_build() {
   CURL_LIB="${CURL}/lib"
   CURL_INCLUDE="${CURL}/include"
 
-  LD_LIBRARY_PATH=${ZLIB_LIB}:${CURL_LIB}
-  ./bootstrap --parallel="$(nproc)" --system-curl -- -DLD_LIBRARY_PATH="${LD_LIBRARY_PATH}" \
+  ./bootstrap --parallel="$(nproc)" --system-curl -- \
     -DZLIB_LIBRARY:FILEPATH="${ZLIB_LIB}/libz.so" -DZLIB_INCLUDE_DIR:PATH="${ZLIB_INCLUDE}" \
     -DCURL_LIBRARY:FILEPATH="${CURL_LIB}/libcurl.so"  -DCURL_INCLUDE_DIR:PATH="${CURL_INCLUDE}"
 

